src/main/java/com/github/manolo8/darkbot/gui/plugins/PluginDisplay.java

```diff
@@ -42,15 +45,20 @@ private JComponent setupUI() {
         return scrollPane;
     }
 
-    private void refreshUI() {
+    void refreshUI() {
         pluginPanel.removeAll();
+        pluginUpdater.clear();
+
+        pluginPanel.add(new PluginUpdateHeader(pluginUpdater, this));
```

Comment:
The update header shouldn't be inside the plugin panel, that's the scroll pane.

Modify setupUI to include:
- `if (pluginPanel != null) return;` 
 - `add(new PluginUpdateHeader(...));`
 - `add(scrollPane)` at the end
 - make the method return void.

Then on line 34 remove the call to "add", as well as the IF, just call setupUI();